pkg/operator/controllers/checker/unsupportedconditionchecker.go

```diff
@@ -0,0 +1,153 @@
+	log *logrus.Entry
+
+	kubernetescli kubernetes.Interface
+	consolecli    consoleclient.Interface
+	configcli     configclient.Interface
+
+	role string
+}
+
+func NewUnsupportedConditionChecker(log *logrus.Entry, kubernetescli kubernetes.Interface, consolecli consoleclient.Interface, configcli configclient.Interface, role string) *UnsupportedConditionChecker {
+	return &UnsupportedConditionChecker{
+		log:           log,
+		kubernetescli: kubernetescli,
+		consolecli:    consolecli,
+		configcli:     configcli,
+		role:          role,
+	}
+}
+
+func (ucc *UnsupportedConditionChecker) Name() string {
+	return "UnsupportedConditionChecker"
+}
+
+func (ucc *UnsupportedConditionChecker) Check(ctx context.Context) error {
+	if ucc.role != operator.RoleMaster {
+		return nil
+	}
+
+	return ucc.checkWorkerNodeCount(ctx)
+}
+
+func (ucc *UnsupportedConditionChecker) checkWorkerNodeCount(ctx context.Context) error {
+	nodes, err := ucc.kubernetescli.CoreV1().Nodes().List(ctx, metav1.ListOptions{
+		LabelSelector: "node-role.kubernetes.io/worker",
+	})
+	if err != nil {
+		return err
+	}
+
+	if len(nodes.Items) < WORKER_NODE_MINIMUM_COUNT {
+		err = ucc.setClusterOperatorStatus(ctx, OperatorUpgradableFalse)
+		if err != nil {
+			return err
+		}
+
+		return ucc.addConsoleBanner(ctx, fmt.Sprintf("Unsupported Cluster State: There needs to be at least %d worker nodes running", WORKER_NODE_MINIMUM_COUNT))
+	}
+
+	err = ucc.setClusterOperatorStatus(ctx, OperatorUpgradableTrue)
+	if err != nil {
+		return err
+	}
+
+	return ucc.consolecli.ConsoleV1().ConsoleNotifications().Delete(ctx, consoleBannerName, metav1.DeleteOptions{})
+}
+
+func (ucc *UnsupportedConditionChecker) setClusterOperatorStatus(ctx context.Context, upgradeableStatus configv1.ConditionStatus) error {
+	co, err := ucc.configcli.ConfigV1().ClusterOperators().Get(ctx, clusterOperatorName, metav1.GetOptions{})
+
+	if err != nil {
+		ucc.log.Errorf("Failed to get the %q operator: %q", clusterOperatorName, err.Error())
+		return err
+	}
+
+	condition := configv1.ClusterOperatorStatusCondition{
+		Type:               configv1.OperatorUpgradeable,
+		Status:             upgradeableStatus,
+		LastTransitionTime: metav1.Now(),
+		Reason:             reasonAsExpected,
+	}
+
+	v1helpers.SetStatusCondition(&co.Status.Conditions, condition)
+
+	_, err = ucc.configcli.ConfigV1().ClusterOperators().UpdateStatus(ctx, co, metav1.UpdateOptions{})
+	if err != nil {
+		ucc.log.Errorf("Failed to update the %q operator: %q", clusterOperatorName, err.Error())
+		return err
+	}
+
+	ucc.log.Infof("Set the %q operator upgradeable status to %q", clusterOperatorName, upgradeableStatus)
+
+	return nil
+}
+
+func (ucc *UnsupportedConditionChecker) addConsoleBanner(ctx context.Context, consoleText string) error {
+	notification, err := ucc.consolecli.ConsoleV1().ConsoleNotifications().Get(ctx, consoleBannerName, metav1.GetOptions{})
+	if err != nil {
+		if !kerrors.IsNotFound(err) {
```

Comment:
Should we be creating the banner when it is not found and not when it is found (with error)?

Right now this reads as "get the banner, if we got an error and that error is not that we couldn't find the banner, create the banner". 

Instead of "get the banner, if we got an error and that error is that we could not find the banner, create the banner"

Reply:
Alternatively, if there is a way to list the banners it might make it more clear when we need to create vs update.

Reply:
@s-amann - Thanks for your time to review and I apologize for my late response. Let me do some more testing and back to your questions then.

Reply:
> Right now this reads as "get the banner, if we got an error and that error is not that we couldn't find the banner, create the banner".

The main logic of this part was referred from`banner.go` which I thought it's the way of convention to `get and create banner` and I just refactored a bit in my case.

Reply:
> Alternatively, if there is a way to list the banners it might make it more clear when we need to create vs update.

The banner name of this checker is hard-coded as a fixed name, I am not sure why it should be listed ?  Please let me know if anything I misunderstood.

Reply:
Listing the banners is a misunderstanding on my part then, I didn't realize there was only one banner. My initial thinking was if we 'list banners' and the result is `[]` then we know we need to create one, but if it came back `[banner-we-want]` then we would know to update. Please disregard this idea if there is only ever one banner and we can determine its name ahead of time.